.github/workflows/test.yml

```diff
@@ -40,7 +40,15 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - run: make docker-coverage
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
+        cache: 'pip'          # caching pip dependencies
+
+    - run: make requirements
+    - run: make requirements-tests
```

Comment:
please. execute `make requirements-tests` before `make requirements`: it's not optimal but this way we are sure that any sub-dependencies version from `requirements-tests` will not be "skewed" from how it should be from `requirements`

Reply:
btw, looking at the errors in the last run I have the impression that every `run` directive is executed in a different "context".
maybe the following?
```
run: make requirements-test && make requirements && make coverage
```

Reply:
As far as I see in the examples for using the [`setup-python`](https://github.com/actions/setup-python#caching-packages-dependencies)

```
steps:
- uses: actions/checkout@v3
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'pip'
    cache-dependency-path: '**/requirements-dev.txt'
- run: pip install -r subdirectory/requirements-dev.txt
```

Maybe the issue could be related to `pip3` used within the `make` goals